post-scan-actions/aws-python-email-notification/handler.py

```diff
@@ -0,0 +1,84 @@
+        # Analyze the message and look for findings
+        message = json.loads(record['Sns']['Message'])         
+        findings = message['scanning_result'].get('Findings')
+
+        if findings:
+            # The email body for recipients with HTML and non-HTML email clients.
+            body_text = '''\
+                CloudOne FSS Email Notification
+                File URL: {file_url}
+                Malware: {malware}
+                Type: {type}\
+                '''.format(file_url=str(message['file_url']),
+                           malware=str(message['scanning_result']['Findings'][0].get('malware')),
+                           type=str(message['scanning_result']['Findings'][0].get('type')))
+            body_html = '''\
+                <html><head></head><body><h1>CloudOne FSS Email Notification</h1><p>
+                <p><b>File URL: </b>{file_url}</p>
+                <p><b>Malware: </b>{malware}</p>
+                <p><b>Type: </b>{type}</p>
+                </html></body></p>\
+                '''.format(file_url=str(message['file_url']),
+                           malware=str(message['scanning_result']['Findings'][0].get('malware')),
+                           type=str(message['scanning_result']['Findings'][0].get('type')))
+
+            # Create a new SES resource and specify a region.
+            client = boto3.client('ses',region_name=aws_region)
+            
+            # Try to send the email.
+            try:
+                response = client.send_email(
+                    Destination={
+                        'ToAddresses': [
+                            recipient,
+                        ],
+                    },
+                    Message={
+                        'Body': {
+                            'Html': {
+                                'Charset': charset,
+                                'Data': body_html,
+                            },
+                            'Text': {
+                                'Charset': charset,
+                                'Data': body_text,
+                            },
+                        },
+                        'Subject': {
+                            'Charset': charset,
+                            'Data': subject,
+                        },
+                    },
+                    Source=sender,
+                )
+            # Display an error if something goes wrong.	
+            except ClientError as e:
+                return {
+                    'statusCode': 200,
+                    'body': json.dumps(e.response['Error']['Message'])
+                }
+                print(e.response['Error']['Message'])
+            else:
+                return {
+                    'statusCode': 200,
+                    'body': json.dumps("Email sent! Message ID:"+response['MessageId'])
+                }
+
+    return {
+        'statusCode': 200,
+        'body': json.dumps("Nothing done.")
+    }
```

Comment:
The return is not used by caller (SNS). Just printing message would be enough.

Reply:
Op

Reply:
Changed